File: adventofcode/01/1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N, sol1, sol2;
	cin >> N;
	vector<int> entries(N);
	for (int i = 0; i < N; i++)
		cin >> entries[i];
	for (int i = 0; i < N; i++) 
		for (int j = i + 1; j < N; j++) {
			if (entries[i] + entries[j] == 2020) 
				sol1 = entries[i] * entries[j];
			for (int k = j + 1; k < N; k++) 
				if (entries[i] + entries[j] + entries[k] == 2020) 
					sol2 = entries[i] * entries[j] * entries[k];
		}
	cout << "Problem1: " << sol1 << endl << "Problem2: " << sol2 << endl;
}

File: adventofcode/2020/01/1.py
n = int(input())
entries = []
for i in range(0, n):
    entries.append(int(input()))
for i in range(0, n):
    for j in range(i+1, n):
        if entries[i] + entries[j] == 2020:
            sol1 = entries[i] * entries[j]
        for k in range(j+1, n):
            if entries[i] + entries[j] + entries[k] == 2020:
                sol2 = entries[i] * entries[j] * entries[k]
print("Problem1: {}\nProblem2: {}".format(sol1, sol2))